tests/testutils.py

```diff
@@ -407,6 +409,62 @@ def skip_if_windows(cls):
     return decorator(cls)
 
 
+def crdb_version(conn, __crdb_version=[]):
+    """
+    Return the CockroachDB version if that's the db being tested, else None.
+
+    Return the number as an integer similar to PQserverVersion: return
+    v20.1.3 as 200103.
+
+    Assume all the connections are on the same db: return a cached result on
+    following calls.
+
+    """
+    if __crdb_version:
+        return __crdb_version[0]
+
+    sver = conn.info.parameter_status("crdb_version")
+    if sver is None:
+        __crdb_version.append(None)
+    else:
+        m = re.search(r"\bv(\d+)\.(\d+)\.(\d+)", sver)
+        if not m:
+            raise ValueError(
+                "can't parse CockroachDB version from %s" % sver)
+
+        ver = int(m.group(1)) * 10000 + int(m.group(2)) * 100 + int(m.group(3))
+        __crdb_version.append(ver)
+
+    return __crdb_version[0]
+
+
+def skip_if_crdb(reason, conn=None):
+    """Skip a test or test class if we are testing against CockroachDB.
+
+    Can be used as a decorator for tests function or classes:
+
+        @skip_if_crdb("my reason")
+        class SomeUnitTest(UnitTest):
+            # ...
+
+    Or as a normal function if the *conn* argument is passed.
+    """
+    if conn is not None:
+        if crdb_version(conn) is not None:
```

Comment:
I wonder if we should also have a decorator that skips based on crdb_version. Something similar to `skip_before_postgres`

Reply:
We can surely make it so. I was waiting to have a use case for it. Do you have it? Is there a feature that you know unsupported in certain CRDB versions and supported in others, which I could test using a docker image?

Reply:
Now it's possible to specify when to skip a test using something like:
```
    @skip_if_crdb("notify", version="<= 20.2")
```
Not used anywhere yet: do you have any place where to use it already?

Reply:
Thanks! One place where it could be used is for enums, which we will only begin supporting in v20.2. However, I don't know if any tests rely on that specifically.